3542/1/lib/gem_manager.rb

```diff
@@ -0,0 +1,42 @@
+require 'yaml'
+require 'terminal-table'
+require_relative 'repo'
+require_relative 'util'
+
+class GemManager
+  def initialize(params)
+    @params = params.map { |param| param.split('=') }.to_h
+    @gem_list = gem_list
```

Comment:
`gem_list` and `top_gems` should be called if flags include `name` or `top`
if they don't this steps are overhead